src/lib/dialog/dialog-ref.ts

```diff
@@ -72,14 +73,15 @@ export class MatDialogRef<T, R = any> {
 
     _overlayRef.keydownEvents()
       .pipe(filter(event => event.keyCode === ESCAPE && !this.disableClose))
-      .subscribe(() => this.close());
+      .subscribe(() => this.close(undefined, 'keyboard'));
   }
 
   /**
    * Close the dialog.
-   * @param dialogResult Optional result to return to the dialog opener.
+   * @param dialogResult Optional result to return to the dialog opener
+   * @param focusOrigin Type of interaction that closed the dialog
    */
-  close(dialogResult?: R): void {
+  close(dialogResult?: R, focusOrigin: FocusOrigin = 'program'): void {
```

Comment:
I think I mentioned this in chat: instead of changing the signature of the `close` method, I think we should add another method, probably internal, like `_closeVia`

It would also make more sense to update the `_closeFocusOrigin` state in such a function rather than changing `_startExitAnimation`

Reply:
Yeah we chatted about that. I just thought that it also would be fine like that, because the second parameter would be optional. Will change it.